### ISSUE TITLE
Option to show only time slots where there are events

### ISSUE THREAD

**Issue by doze** [fullcalendar/fullcalendar#5429]
I'd like to create a calendar which displays google calendar events on a web page. In addition, I'd like that in timegrid view, only the times which have events would be visible.

Currently it's not very easy to achieve this as setting minTime and maxTime causes events to be refetched (#4698). So cannot set those options in eventSourceSuccess. But even if that issue would be fixed I'm not sure if setting minTime and maxTime there would do it, because the eventSourceSuccess is not triggered for example when user switches from week to day view. But the minTime and maxTime should need to be updated in that case also and have the grid re-rendered with new values. And also when the "navlinks" are used, eg. day header clicked, that event is not triggered. Maybe more cases also, but those atleast.

So ultimately, option to make this automatic would be nice!

**Comment by acerix** [comment]
Do you mean having `minTime` and `maxTime` update automatically to fit all the events in the day, but not hiding slots in the middle of the day?

I believe what you suggested would work by turning `lazyFetching` off so `eventSourceSuccess` is always triggered.

**Comment by doze** [comment]
Yes, that's what I mean. It would of course be nice to not have events fetched too often because of this, but retrieve once and work with them in memory. 

I have currently implemented this in very hacky way by setting onclick event handlers for the various navigation buttons and changing minTime and maxTime there (see https://stackoverflow.com/a/61891841/2113540), but by doing so, the event's gets retrieved multiple times for the same timespan (first by me when getting the min/max times and then by the #4698 bug) and this causes delay when switching weeks in the calendar for example. I would need to do some kind of caching to the google calender event fetcher... so it would not retrieve events for the same time period multiple times when called in sort time span.
